pkg/core/chain/chain.go

```diff
@@ -200,17 +202,26 @@ func New(ctx context.Context, eventBus *eventbus.EventBus, rpcBus *rpcbus.RPCBus
 	}
 
 	// Hook the chain up to the required topics
-	cbListener := eventbus.NewCallbackListener(chain.onAcceptBlock)
-	eventBus.Subscribe(topics.Block, cbListener)
-	initListener := eventbus.NewCallbackListener(chain.onInitialization)
-	eventBus.Subscribe(topics.Initialization, initListener)
+	chain.blockChan = make(chan message.Message, 100)
+	eventBus.Subscribe(topics.Block, eventbus.NewChanListener(chain.blockChan))
+
+	chain.initializationChan = make(chan message.Message, 1)
+	eventBus.Subscribe(topics.Initialization, eventbus.NewChanListener(chain.initializationChan))
 	return chain, nil
 }
 
 // Listen to the collectors
 func (c *Chain) Listen() {
 	for {
 		select {
+		case m := <-c.blockChan:
+			if err := c.onAcceptBlock(m); err != nil {
+				log.WithError(err).Warnf("Handling %s failed", topics.Block.String())
```

Comment:
could you please log it without Warnf and split the fields? that helps with the monitoring.

Reply:
No, this still can be easily parsed by monitoring.